lib/netext/httpext/error_codes.go

```diff
@@ -109,7 +109,13 @@ func http2ErrCodeOffset(code http2.ErrCode) errCode {
 
 // errorCodeForError returns the errorCode and a specific error message for given error.
 func errorCodeForError(err error) (errCode, string) {
-	switch e := errors.Cause(err).(type) {
+	// We explicitly check for `Unwrap()` in the default switch branch, but
+	// checking for the concrete error types first gives us the opportunity to
+	// also directly detect high-level errors, if we need to, even if they wrap
+	// a low level error inside.
```

Comment:
I would prefer if you port the tests as well before we merge this.

The logic in #1660 is the reverse: we first go for the most inner error and then upwards while here we go from the outer one inner. 
The current code also in practice just goes for the inner one, *only*, unless it is url.Error in which case it goes for the inner error (this probably is not needed?!?). 

Either way the tests will show whether at least for those cases this keeps working and they are perfectly fine in a separate commit in this PR

Reply:
:+1: I'll port the tests as well as some of your other changes from that PR. 

My logic with the current code is that, if you want to, you can always drill down and get a wrapped error, even before reaching the `default`, as with `url.Error`. But it doesn't _force_ you to do so, you can catch top-level errors even if they wrap another one, if that's what's necessary to return the proper error code.

Reply:
I ported your tests and some of the fixes from https://github.com/k6io/k6/pull/1660, then fixed the one broken test with t he new logic. Finally, I fixed some lint issues and and refactored the whole `errorCodeForError()` function to be a bit more readable by splitting apart the `*net.OpError` detection in a new function `errorCodeForNetOpError()`. 

Please read commit by commit, and if you're satisfied everything usable from https://github.com/k6io/k6/pull/1660 has been ported, please close that PR.